### PR TITLE
Add tests for (real world) package installation

### PR DESCRIPTION
**Pull request by csjh**
### Description

Adds an e2e test that packs every package into tarballs and installs from those using npm

### CODE REVIEW

**Comment by csjh** [comment]
Not sure where to put the `node ./build-deps-to-tgz.js && npm i` step so it's in sources for now

**Comment by csjh** [comment]
And need to move away from `import("fs").glob` for the node 18 suite

**Comment by csjh** [comment]
`e2e/packaging` tests are expected to fail until #2866 merges

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -65,9 +65,9 @@ jobs:
       - name: Install Playwright and browsers
         run: pnpm playwright install --with-deps
 
-      - name: Run sources
+      - name: Run setup
```
Comment (zachstence) [inline .github/workflows/e2e.yml:68]: I'd rather not modify all the `e2e` projects to have a `setup` script.

Instead, can you add another step to this workflow to run `setup` if it exists?

```sh
pnpm run --if-present setup
```

File: package.json [hunk 2]
```diff
@@ -66,7 +66,8 @@
 		"unified": "9.1.0",
 		"unist-util-visit": "4.1.2",
 		"uvu": "0.5.2",
-		"vite": "5.4.11"
+		"vite": "5.4.11",
+		"glob": "11.0.0"
```
Comment (zachstence) [inline package.json:70]: Install this as a `devDependency` in `e2e/package.json` instead of the root `package.json`

File: pnpm-workspace.yaml [hunk 3]
```diff
@@ -1,5 +1,6 @@
 packages:
   - 'e2e'
   - 'e2e/*'
+  - '!e2e/packaging'
```
Comment (zachstence) [inline pnpm-workspace.yaml:4]: Add a comment why this is excluded

**Review by zachstence — CHANGES_REQUESTED** [review]

**Pull request merged**